network/p2p/p2pbuilder/libp2pNodeBuilder.go

```diff
@@ -413,18 +398,19 @@ func (builder *LibP2PNodeBuilder) Build() (p2p.LibP2PNode, error) {
 					ctx.Throw(fmt.Errorf("could not stop libp2p node: %w", err))
 				}
 			}
-		})
-	cm = cm.AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
-		if builder.gossipSubTracer == nil {
-			builder.logger.Warn().Msg("libp2p tracer is not set")
-			ready()
-			return
-		}
+		}).
+		AddWorker(func(ctx irrecoverable.SignalerContext, ready component.ReadyFunc) {
+			// encapsulates the startup logic for the libp2p node.
```

Comment:
not a change from this PR, but on line 412, should we wait for `<-builder.gossipSubTracer.Done()`?

Reply:
https://github.com/onflow/flow-go/commit/0600d5c82f5a8ddeb311911a316826b08b0ac514